dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/MetadataDumper.java

```diff
@@ -315,23 +315,25 @@ private File getOutputFile(Connector connector, ConnectorArguments arguments) {
     String defaultFileName = connector.getDefaultFileName(arguments.isAssessment());
     return arguments
         .getOutputFile()
-        .map(
-            file -> {
-              String fileName = file.getPath();
-              if (StringUtils.endsWithIgnoreCase(fileName, ".zip")) return file;
-              if (file.isFile()) {
-                throw new IllegalStateException(
-                    String.format(
-                        "A file already exists at %1$s. "
-                            + "If you want to create a directory, please provide the path to the directory. "
-                            + "If you want to create %1$s.zip, please add the `.zip` extension manually.",
-                        fileName));
-              }
-              return new File(file, defaultFileName);
-            })
+        .map(file -> getVerifiedFile(defaultFileName, file))
         .orElseGet(() -> new File(defaultFileName));
   }
 
+  private File getVerifiedFile(String defaultFileName, File file) {
+    String fileName = file.getPath();
+    boolean isZipFile = StringUtils.endsWithIgnoreCase(fileName, ".zip");
+
+    if (file.isFile() && !isZipFile)
```

Comment:
Can we get some braces on this `if` please? I'm usually fine with omitting braces on short `if` bodies, but this one is long enough that I legitimately got confused by what the `return` was doing the first time I read the method until I realized it wasn't inside the `if`.